src/libraries/perceptiveModels/src/springyFingers.cpp

```diff
@@ -293,8 +293,8 @@ bool SpringyFingersModel::fromProperty(const Property &options)
     if (!driver.open(prop))
         return false;
     
-    port->open("/"+name+"/"+part_analog+"/analog:i");
-    string analogPortName("/"+robot+"/"+part_analog+"/analog:o");
+    port->open("/"+name+"/"+part_analog+"/MAIS");
+    string analogPortName("/"+robot+"/"+part_analog+"/MAIS");
```

Comment:
`perceptiveModels` is quite old and makes use of direct access to ports. Replacing port direct access with a device driver is not doable because of the internal machinery.

However, it should be still ok this way, @Nicogene.

Reply:
My doubt is that if this line is still working
https://github.com/robotology/icub-main/blob/bd74c24d0aae5918d62a97ad3354ed5793a0b0f7/src/libraries/perceptiveModels/src/sensors.cpp#L91

Because the protocol of `AnalogServer` and `MultipleAnalogSensorsServer` is different, it is expecting N doubles

https://github.com/robotology/icub-main/blob/bd74c24d0aae5918d62a97ad3354ed5793a0b0f7/src/libraries/perceptiveModels/src/ports.cpp#L47

Instead now the measure is nested in Bottles (cc @traversaro)

Reply:
Got it 👍🏻 
I'll see what I can do with the abstract port class of that library.

Reply:
In theory we can also read directly via the port by generating the message in thrift, I guess YARP installs it and we can generate the code locally to use.

Reply:
Actually, I can do something specific for `yarp::dev::IEncoderArrays` starting from

https://github.com/robotology/icub-main/blob/bd74c24d0aae5918d62a97ad3354ed5793a0b0f7/src/libraries/perceptiveModels/include/iCub/perception/sensors.h#L166-L192